app/javascript/components/Churn/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { router } from "@inertiajs/react";
+import { lightFormat } from "date-fns";
+import * as React from "react";
+
+export type ChurnData = {
+  start_date: string;
+  end_date: string;
+  metrics: {
+    customer_churn_rate: number;
+    last_period_churn_rate: number;
+    churned_subscribers: number;
+    churned_mrr_cents: number;
+  };
+  daily_data: {
+    date: string;
+    customer_churn_rate: number;
+    churned_subscribers: number;
+    churned_mrr_cents: number;
+  }[];
+};
+
+import { AnalyticsLayout } from "$app/components/Analytics/AnalyticsLayout";
+import { ProductsPopover } from "$app/components/Analytics/ProductsPopover";
+import { useAnalyticsDateRange } from "$app/components/Analytics/useAnalyticsDateRange";
+import { ChurnChart } from "$app/components/Churn/ChurnChart";
+import { ChurnDateRangePicker } from "$app/components/Churn/ChurnDateRangePicker";
+import ChurnQuickStats from "$app/components/Churn/ChurnQuickStats";
+import { Progress } from "$app/components/Progress";
+
+import placeholder from "$assets/images/placeholders/sales.png";
+
+export type Product = {
+  id: string;
+  name: string;
+  unique_permalink: string;
+  alive: boolean;
+};
+
+export type ChurnProps = {
+  churn_props: {
+    has_subscription_products: boolean;
+    products: Product[];
+  };
+  churn_data: ChurnData | null;
+};
+
+const isChurnData = (data: unknown): data is ChurnData =>
+  typeof data === "object" &&
+  data !== null &&
+  "daily_data" in data &&
+  "metrics" in data &&
+  "start_date" in data &&
+  "end_date" in data;
+
+const Churn = ({ churn_props, churn_data }: ChurnProps) => {
+  const { has_subscription_products, products: initialProducts } = churn_props;
+  const dateRange = useAnalyticsDateRange();
+  const [data, setData] = React.useState<ChurnData | null>(churn_data || null);
+  const [productOptions, setProductOptions] = React.useState(
+    initialProducts.map((product) => ({ ...product, selected: product.alive })),
+  );
```

Comment:
addressing your previous comment here, selected attribute is needed since this data is being passed to ProductsPopover component and it needs to know which products are selected and initially we select all alive/active products